power_grid_model_c/power_grid_model/include/power_grid_model/sparse_mapping.hpp

```diff
@@ -54,31 +54,92 @@ inline SparseMapping build_sparse_mapping(IdxVector const& idx_B_in_A, Idx const
     auto const n_A = static_cast<Idx>(idx_B_in_A.size());
     using SparseEntry = std::pair<Idx, Idx>;
     std::vector<SparseEntry> entries(n_A);
+
     std::transform(idx_B_in_A.cbegin(), idx_B_in_A.cend(), IdxCount{0}, entries.begin(), [](Idx j_B, Idx i_A) {
         return SparseEntry{i_A, j_B};
     });
-    // initialize result
+
     SparseMapping sparse_mapping;
     sparse_mapping.indptr.resize(n_B + 1);
     sparse_mapping.reorder.resize(n_A);
-    // counting sort column, keep row as sorted
+
     IdxVector counter(n_B, 0);
     for (auto const& entry : entries) {
         ++counter[entry.second];
     }
-    // accumulate value to indptr
+
     std::inclusive_scan(counter.cbegin(), counter.cend(), sparse_mapping.indptr.begin() + 1);
-    // copy back
+
     std::copy(sparse_mapping.indptr.cbegin() + 1, sparse_mapping.indptr.cend(), counter.begin());
-    // sort
+
     for (auto it_entry = entries.crbegin(); it_entry != entries.crend(); ++it_entry) {
         sparse_mapping.reorder[--counter[it_entry->second]] = it_entry->first;
     }
+
     assert(sparse_mapping.indptr[0] == 0);
     assert(sparse_mapping.indptr.back() == n_A);
     return sparse_mapping;
 }
 
-} // namespace power_grid_model
+struct DenseMapping {
+    IdxVector indvector;
+    IdxVector reorder;
+};
+
+inline DenseMapping build_dense_mapping(IdxVector const& idx_B_in_A, Idx const n_B) {
+    auto const n_A = static_cast<Idx>(idx_B_in_A.size());
+
+    using DenseEntry = std::pair<Idx, Idx>;
+    DenseMapping dense_mapping;
+
+    if (static_cast<double>(n_A) * log(static_cast<double>(n_A)) <=
+        static_cast<double>(n_A) + static_cast<double>(n_B)) {
```

Comment:
this is unreadable. please replace with something like

```suggestion
    auto const comparison_sort_complexity = [](Idx n_a, Idx /* n_B */) {
        auto const double_n_a = static_cast<double>(n_a);
        return double_n_a * std::log(double_n_a);
    };
    auto const counting_sort_complexity = [](Idx n_a, Idx n_b) { return static_cast<double>(n_a + n_b); };
    if (comparison_sort_complexity(n_A, n_B) <= counting_sort_complexity(n_A, n_B)) {
```

or (without lambdas)

```suggestion
    const auto comparison_sort_complexity = static_cast<double>(n_A) * std::log(static_cast<double>(n_B));
    const auto counting_sort_complexity = static_cast<double>(n_A + n_B);
    if (comparison_sort_complexity < counting_sort_complexity) {
```